app/src/main/java/de/danoeh/antennapod/preferences/PreferenceController.java

```diff
@@ -227,6 +227,33 @@ private void setupInterfaceScreen() {
                     return true;
                 });
 
+        ui.findPreference(UserPreferences.PREF_BACK_BUTTON_BEHAVIOR)
+                .setOnPreferenceChangeListener((preference, newValue) -> {
+                        if(newValue.equals("page")) {
+                            final Context context = ui.getActivity();
+                            final String[] navTitles = context.getResources().getStringArray(R.array.back_button_go_to_pages);
+                            final int choice[] = { UserPreferences.getBackButtonGoToPage() };
+
+                            AlertDialog.Builder builder = new AlertDialog.Builder(context);
+                            builder.setTitle(R.string.back_button_go_to_page_title);
+                            builder.setSingleChoiceItems(navTitles, choice[0], (dialogInterface, i) -> {
+                                if(i >= 0) {
+                                    choice[0] = i;
+                                }
+                            });
+                            builder.setPositiveButton(R.string.confirm_label, (dialogInterface, i) -> {
+                                if (choice[0] != UserPreferences.getBackButtonGoToPage()) {
+                                    UserPreferences.setBackButtonGoToPage(choice[0]);
```

Comment:
I would save the tag of the fragment instead of the index. That way, you could prevent the code duplication for each fragment in MainActivity. Additionally, this allows re-ordering the list without breaking existing user's data.